packages/ethereum-contracts/contracts/Contract_Example.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import "./EIP712_Canvas.sol";
+
+string constant topic = "xyz.canvas.myapp";
+
+contract Contract_Example {
+
+
+  mapping(bytes32 => bool) public appliedActionHashes;
+  uint256 public upvotes;
+
+  function claimUpvoted(
+    address expectedAddress,
+    bytes memory sessionSignature,
```

Comment:
Yes if the `Session` object has an `authorizationData` field

Reply:
So we could also write this to expect the `Session` struct to have an `authorizationData` field, then use the signature inside authorizationData instead of passing the signature in separately. This is correct I think, it would just be mirroring what `verifySession` does.